code/game/objects/structures/mirror.dm

```diff
@@ -207,3 +207,28 @@
 /obj/structure/mirror/kick_act()
 	..()
 	shatter()
+
+/obj/structure/mirror/magic
+	name = "magic mirror"
+	desc = "Mirror mirror on the wall, who's the most powerful of them all? It hums with arcane power."
+	icon_state = "mirrormagic"
+
+/obj/structure/mirror/magic/attack_hand(mob/M)
+	if(!shattered)
+		var/which = alert(M, "What would you like to change?", "Magic Mirror", "Name", "Gender", "Appearance")
```

Comment:
Needs a cancel button

Reply:
Alert won't allow more than three buttons. Is there another way to make it able to be cancelled?

Reply:
Instead of using an alert, use an input:
`var/which = input("Change what?", "Magic Mirror") as null|anything in list("Name","Gender","Appearance")`